### PR TITLE
Outfit wars support

### PR DESCRIPTION
**Pull request by Maelstromeous**
Adds support for outfit wars.

## Tasks

- [x] Create new section specific for just outfit wars, showing:
  - [x] List of all matches, seperated by server
  - Global statistics with options to filter by server @Maelstromeous 
    - [ ] Players
    - [ ] Outfits
    - [ ] Weapons
    - [ ] Vehicles
    - [ ] Facilities
    - [ ] Ensure API returns purely statistics from LIVE metagame on live metagame pages
  - [ ] `In progress` Look into creating a "tournament" style listing showing all outfits and their matches, broken down into rounds like Challonge
- [x] Adapt current alert page / create a new one that pulls in the statistics for the instance @ryanjsims 
  - Could possibly get away with using the current _alert.vue page for this with adaptations based on the eventType that's passed
- [x] Add RTM support
- [x] Hide bracket calculations graph as it makes no sense @Maelstromeous 
- [x] Hide any mention of brackets @Maelstromeous 
- [x] Player / Outfit stats: Show teams (not faction) for each player with some means of showing the faction (maybe)? @Maelstromeous 
- [x] Population History graph needs OW awareness to display team populations rather than faction @Maelstromeous 
- [x] BRs need to show teams not factions @Maelstromeous 
- [x] Combat History graph is not getting kill data, 
  - [x] and needs to show teams not factions @Maelstromeous 
- [x] Merge latest fixes to weapon names into this branch @ryanjsims 
- [ ] Make alert capture timeline always show outfit names even if no outfit id was sent with the facility control event @ryanjsims 
- [ ] Fix match list page filtering
- [ ] Add match list page filter by outfit name/tag/faction
- [ ] Remove match list page filter by team
- [ ] Ranking page: add actual ranks somewhere
- [ ] Ranking page: Reduce dead vertical space
- [ ] Ranking page: outfit arrows should take you to a filtered by outfit match page

Polish:
- [ ] Add custom OG image for outfit wars matches

### CODE REVIEW

File: components/RTM/RealTimeMonitor.vue [hunk 1]
```diff
@@ -33,6 +33,24 @@
...
+            class="p-1 border-b border-gray-500 border-no-bottom"
+          >
+            <RealTimeOutfitWar
+              :world="outfitwar.world"
+              :zone="outfitwar.zone"
+              :time-started="outfitwar.timeStarted"
+              :duration="outfitwar.duration"
+              :result="outfitwar.result"
+              :instance-id="outfitwar.instanceId"
+              :pops="getPops(outfitwar.instanceId)"
```
Comment (Maelstromeous) [inline components/RTM/RealTimeMonitor.vue:49]: This likely won't work at all yet, aggregator needs changing to detect these pops on the instanced zones

File: components/alert/AlertResult.vue [hunk 2]
```diff
@@ -113,6 +130,29 @@ export default Vue.extend({
...
       }
     },
+    roundByPhase(): string {
+      if (this.alert.ps2alertsEventType !== Ps2alertsEventType.OUTFIT_WARS_AUG_2022) {
+        return ''
+      }
+      const match = this.alert as InstanceOutfitWarsResponseInterface;
+      if (match.phase === Phase.QUALIFIERS) {
+        return match.round.toString()
+      } else if (match.phase === Phase.PLAYOFFS) {
```
Comment (Maelstromeous) [inline components/alert/AlertResult.vue:140]: I'm a bit confused by this - is it spitting out the round number as a string here, or is there other magic going on that translates it to say "Playoffs" or "Qualifiers"?

File: components/outfitwars/MatchEntry.vue [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+      if (this.match.result.draw) {
+        return {
+          'bg-tint-light': true,
+          'border-other-alt': true,
+        }
+      }
+
+      return {
+        ...FactionBgClass(this.match.result.victor),
```
Comment (Maelstromeous) [inline components/outfitwars/MatchEntry.vue:178]: This method needs updating to support Teams

File: interfaces/instance-entries/OWMapControlInterface.ts [hunk 4]
```diff
@@ -0,0 +1,6 @@
+export interface OWMapControlInterface {
```
Comment (Maelstromeous) [inline interfaces/instance-entries/OWMapControlInterface.ts:1]: [This probably should be using ](https://github.com/ps2alerts/constants/blob/outfit-wars/interfaces/OutfitwarsTerritoryResultInterface.ts)

File: components/alert/AlertResult.vue (hunk 2 above)
Comment (ryanjsims) [reply components/alert/AlertResult.vue:140]: Its spitting out the round number as a string - I made a new filter for that to make it more clear

**Comment by ryanjsims** [comment]
Additional outstanding items to fix:

[] Population History graph needs OW awareness to display team populations rather than faction
[] Combat History graph is not getting kill data, and will probably need the same OW awareness as population history

**Comment by ryanjsims** [comment]
- [ ] Make alert capture timeline always show outfit names even if no outfit id was sent with the facility control event

**Pull request merged**